### DISCUSSION TITLE
Modify Camera Position in Scene

### DISCUSSION THREAD

**Discussion by kunal-rp** [AR-js-org/AR.js#495] [Help]
Is it possible to modify the [perspective camera's ](https://threejs.org/docs/#api/en/cameras/PerspectiveCamera) position within the scene ? 

For my use case, I would like to set the z position to the middle of the near and far. The issue I am running into is that the marker positions are too small for me to scale down certain text and mesh sizing/coordinates.

Moreover marker/nft position detection should updated as well since the coordinate scale has increased

**Comment by nickw1** [comment]
I don't think you can do exactly that (as by definition, the camera is the current viewpoint) but you might be able to improve the effect by changing the camera's parameters.

Or create a secondary camera in front of the main one, as a child of it (with a given `z` displacement) and allow the user to swap between the two?

Apologies if I've misunderstood what you're trying to do.

**Reply by kunal-rp** [reply]
What I'm trying to understand is how markers detected in the camera feeds/source video is mapped into a threejs scene. 

Specifically how are the coordinates determined ? I initially thought that depth in detection is mapped to the camera's z position depending on the scale (i.e.- camera's position is initially set to 5, so range would be 5 - far value ) , but it seems like modification of the camera does not adjust the detection scale. This same problem occurs for x & y coordinates.

**Comment by nickw1** [comment]
Ah, ok, I may have to leave that to others e.g. @kalwalt to answer that when they are able to. I am not au-fait with the marker side of AR.js, only the location-based side.

**Reply by kunal-rp** [reply]
bump on this

**Comment by kalwalt** [comment]
@kunal-rp i think modifying the camera require also modify the Artoolkit5 setup for the camera, If you are using marker or NFT this is done initially and not sure if it can be changed after, without restarting the whole app. The [loadCamera](https://github.com/AR-js-org/artoolkit5-js/blob/3f4f24603e7b19385c26d7d6006f74abe8cd70c2/src/ARToolkit.js#L84-L103) function load a generic parameter file with the camera settings (near, far, distortion factors...) this is required to generate the projection Camera matrix used by the Threejs camera. In another words we create a camera aligned with the camera used by the Emscripten code. You can see this in ARjs Context:

https://github.com/AR-js-org/AR.js/blob/be7c9083df5a651c2ab978ddd7c1a33409c4d40b/three.js/src/threex/arjs-context.js#L329-L346

**Reply by kunal-rp** [reply]
Could you expand on the required changes in modifying the artoolkit setup? would this require a custom calibration, or would it be possible to use the default params and modify the depth through the camera matrix ( during toolkit creation and setup in app )

**Reply by kunal-rp** [reply]
bump
